### ISSUE TITLE
Firefox Webextension

### ISSUE THREAD

**Issue by engelju** [babyman/quick-tabs-chrome-extension#189]
Hi! I'm a really big fan of your extension, and use it daily. Lately though I've been transitioning to Firefox and was wondering if you'd be open to port this.

Since Firefox 57 I believe, Firefox enforces a new Addon-Technology called "WebExtensions" which basically build upon chrome extensions. So it should be relatively easy to do.

A few helpful links I've found:
- [A Step-by-Step Guide of how RES was ported from Chrome to a FF WebExt](https://hacks.mozilla.org/2015/10/porting-chrome-extensions-to-firefox-with-webextensions/)
- [Official FF Docs on How to Port](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Porting_a_Google_Chrome_extension)
- [Let Mozilla check if your Chrome extension is base-compatible with WebExt](https://www.extensiontest.com/)
- [Packaging an extension folder into a .CRX file](https://developer.chrome.com/extensions/packaging) (needed for extensiontest.com)
- [Temporarly install the extension in Nightly](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Temporary_Installation_in_Firefox)

What I did so far:
- cloned this repo
- packaged the extension into a .csr
- run the csr agains extensiontest.com, which yielded: https://www.extensiontest.com/test/28052fb0-92f9-11e7-97dd-c97ef306ba23
```
{
  "compat": [],
  "errors": [
    {
      "message": "Banned 3rd-party JS library",
      "locations": [
        {
          "file": "js/jquery-1.11.2.min.js"
        }
      ]
    }
  ],
  "warnings": [
    {
      "message": "Inline scripts blocked by default",
      "locations": [
        {
          "file": "popup.html"
        }
      ]
    },
    {
      "message": "Unsafe assignment to innerHTML",
      "locations": [
        {
          "file": "popup.js",
          "line": 463
        }
      ]
    }
  ]
}
```
- tried the extension in Nightly, and it mostly worked (he keybindings wouldn't work and the numbers icon wouldn't reload)

I'd appreciate it, if someone other would take a look at this, and if I have some more time in the future, maybe I'd even spend some more time on it.

**Comment by hackel** [comment]
I would love to be able to use this in Firefox as a replacement for my beloved "All Tabs Helper."  I switch between browsers frequently and the consistency would be much appreciated.  Now that FF57 is in beta, I'm even more desperate to find solid replacements like this.  It should also work in Vivaldi, Opera, etc.

**Comment by ErichDonGubler** [comment]
If somebody could mentor me in making a Firefox extension, I'm more than happy to try to put this over the line.

**Comment by 3v1n0** [comment]
Not really hard: https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Porting_a_Google_Chrome_extension

**Comment by engelju** [comment]
A working alternative is [Most-Recent-Tab](https://github.com/thePaulV/Most-Recent-Tab), which uses the shortcut `Control-shift-1` on Windows and `Command-shift-1` on Mac.

[Since the shortcut can't be customized at this very moment](https://bugzilla.mozilla.org/show_bug.cgi?id=1303384), I created [a fork](https://github.com/engelju/Most-Recent-Tab) to change the shortcut to `Ctrl+Shift+E`, which is a bit closer to the shortcut this very extension was using.

**Comment by ErichDonGubler** [comment]
@engelju The alternative you've provided seems to have a very limited set of features compared to this extension -- it seems the only functionality it offers is navigating to the last-used tab?

**Comment by engelju** [comment]
@ErichDonGubler yup, sorry I should have clarified that. The linked web-extension only provides a shortcut for the most recent tab.

**Comment by omares** [comment]
While searching for an alternative to use with firefox, i stumbled across [tab-search](https://github.com/reblws/tab-search).

It should cover most of the basic needs quick-tabs provides.

**Comment by 3v1n0** [comment]
[saka](https://addons.mozilla.org/en-US/firefox/addon/saka/) is also pretty nice, and having it on ctrl+space is quite handy.

**Comment by hackel** [comment]
If anyone is searching for a good solution for Firefox, the original [All Tabs Helper](https://addons.mozilla.org/en-US/firefox/addon/all-tabs-helper/) has now been ported to a WebExtension.  Perhaps soon it will even come to Chrome!

**Comment by leedoyle** [comment]
@hackel @3v1n0  @omares thanks! I'll check it out. 
However, I still hope Quick Tabs will get ported to FX.

**Comment by district10** [comment]
As an alternative, I use [Vim Vixen](https://addons.mozilla.org/en-US/firefox/addon/vim-vixen/)'s buffer feature to search and nav to specific tab: (press `b` then type)

![image](https://user-images.githubusercontent.com/5262037/37248137-414c39a8-2504-11e8-8406-5a2e4bc20b31.png)

It works. But not good. Hope to see Quick Tabs ported to FireFox one day.

**Comment by cwagner22** [comment]
[TabSearch](https://addons.mozilla.org/en-US/firefox/addon/tab_search/) is great too.

**Comment by poige** [comment]
While there could be alternatives I'd prefer to have this one to be ported as wide as possible. Say, Opera and Firefox.

Moreover, since Quick Tabs is a _great time saver_ to me (thank you) and anyways I was thinking to donate just out of gratitude I'd gladly pay a contribution for this work to be done. And if others interested users would join it could be some money and hopefully enough money to have it done. ;-)

**Comment by Ulyssys** [comment]
Just voicing my support for a Firefox port. The UI and behavior of this extension (on Chrome) is preferred to me over some of the alternatives I've looked at.

**Comment by vancetran** [comment]
Here's an un-official port of Quick Tabs for Firefox. Hoping for the official one soon! It's literally the one extension keeping me on Chrome.
https://addons.mozilla.org/en-US/firefox/addon/quick-tabs-ported/

**Comment by ErichDonGubler** [comment]
@vancetran: That port works wonderfully. Thank you for linking it!

**Comment by kyle0r** [comment]
> 
> 
> Here's an un-official port of Quick Tabs for Firefox. Hoping for the official one soon! It's literally the one extension keeping me on Chrome.
> https://addons.mozilla.org/en-US/firefox/addon/quick-tabs-ported/

I've been using this FF ext few months, as I'm moving away from Chrome back to FF. Its stable.

I'd volunteer someone with free time 😜 to collaborate with the dev, who [from the FF plugin ratings comments](https://addons.mozilla.org/en-US/firefox/addon/quick-tabs-ported/reviews/1479180/) seems open enough, to fork this repo and paste/merge the current FF addon code into the fork. The fork can then be easily maintained and kept up-to-date etc.

**Comment by ErichDonGubler** [comment]
Alright, I'll be *that guy* and bump this. This is a fantastic extension, and it's a shame that Firefox users don't have a first-class experience like what this extension provides yet. The [ported clone](https://github.com/babyman/quick-tabs-chrome-extension/issues/189#issuecomment-497402949) seems to work fine, so I don't see a reason that this can't be done straightforwardly. What I want to know is this: **are maintainers open to merging changes for Firefox compatibility?** I'd be more than happy to take initiative if that's the case -- but I don't want to waste time, having never needed to maintain an FF extension before.

@babyman, if you're not interested in tackling this issue, that's fine. This is an open-source world, and your time and energy are finite. It'd sure be nice if you'd let us know, though!

**Comment by babyman** [comment]
It's all good, thanks for being **that guy** :).  I really just want to clean up the stale tickets to leave things open that are still relevant and of interest to everyone.  I'm going to flag this as a feature request to keep it open and prevent the bot picking it up again.

**Comment by 3v1n0** [comment]
Is there a fork for the firefox code?

Although I'd really prefer to have this merged in this repo...
In case is not, would be nice to support also keybindings to move around the view (like to choose the selected item).

**Comment by babyman** [comment]
There's no fork for FireFox, ideally it would work using the standard APIs but I'll take a look and see what's involved.

I haven't used FF in ages but I hear it's making a strong comeback.

**Comment by ErichDonGubler** [comment]
@3v1n0 @babyman: One can inspect the source code of the [ported clone for Firefox]](https://github.com/babyman/quick-tabs-chrome-extension/issues/189#issuecomment-497402949) by version [here](https://addons.mozilla.org/en-US/firefox/addon/quick-tabs-ported/versions/), but there's no Git database in the archives (I ran `unzip -l quick_tabs_ported-1.0.6-fx.xpi` from a *nix terminal), which means some work would have to be done to figure out what revision the porting changes were applied to. Annoying, though not am impossible task.

**Comment by ErichDonGubler** [comment]
After doing some digging, the last commit in this repo's history that could be the base version of the ported extension is 0c13cdf1a0d6f5f96adbfe0c84f8c88e2f09d461, which I've marked in my fork under the [`maybe-port-fork-point`](https://github.com/ErichDonGubler/quick-tabs-chrome-extension/tree/maybe-port-fork-point) branch. Most of the diffs aren't outrageous, but I wonder if there's a better match for those the base commit of those changes.

**Comment by a-pav** [comment]
Please port it for use in firefox. There's no decent alternative to this extension on the entire addons.mozilla.org. There's TabSearch but it still falls short on the feature list, functionality and, honestly, doesn't look good either.

**Comment by DesertBanana** [comment]
@vancetran @ErichDonGubler @kyle0r @3v1n0 @a-pav 

Are you all still using the unofficial port that vancetran linked?

I was using it fine for years and then randomly it stopped giving keyboard focus to the search box upon activation via keyboard shortcut. This makes it pretty clunky to use now. I did test it on a few fresh Firefox installs and different computers, but I'm wondering if any of you are having the same issue.

**Comment by a-pav** [comment]
> Are you all still using the unofficial port that vancetran linked?

I never used that port.
I've settled with **TabSearch**  which is still working fine to this day despite not being updated for the past 6 years or so. At some point I have also installed **Sideberry**, though it's not perfect either and I don't use it as often as I use **TabSearch**.

Anyway, I think I will be able to remove them both if ever **Quick Tabs** made an official Firefox port.

**Comment by ErichDonGubler** [comment]
@DesertBanana: 

> I was using it fine for years and then randomly it stopped giving keyboard focus to the search box upon activation via keyboard shortcut. This makes it pretty clunky to use now. I did test it on a few fresh Firefox installs and different computers, but I'm wondering if any of you are having the same issue.

I run into the same issue, and it has caused me to abandon what was otherwise a solid flow.  🫤 I have recently adopted a tab tree extension ([Sidebery](https://github.com/mbnuqw/sidebery) too!), and that's served a lot of the use case I originally had for fuzzy searching, but it's not a perfect replacement.

**Comment by kyle0r** [comment]
@DesertBanana 

> @vancetran @ErichDonGubler @kyle0r @3v1n0 @a-pav
> 
> Are you all still using the unofficial port that vancetran linked?
> 
> I was using it fine for years and then randomly it stopped giving keyboard focus to the search box upon activation via keyboard shortcut. This makes it pretty clunky to use now. I did test it on a few fresh Firefox installs and different computers, but I'm wondering if any of you are having the same issue.

I'm still using the Firefox "quick-tabs-ported" extension. A bug has developed with a recent Firefox release - like you mentioned with the search field focus issue. For me the reproduction of the bug was intermittent, most of the time it **does not** focus on the search field when the popup opens, sometimes it does. I've noted that mouse clicking on the toolbar button does seem to focus on the search field whereby keyboard shortcut activation invariably does not focus on the search field.

If you look at the comments on the extension page, others users have also posted snippets of fixes/tweaks for other topics. This motivated me to see if I could fix the focus issue, which I think I have been able to do.

One can inspect the extension code via the Firefox developer tools. Info on how to debug `popup.html` on superuser [here](https://superuser.com/a/1743236/59966). Once you're debugging `popup.html` you can check out `popup.js` on the **debugger** tab.  
**Unfortunately** it looks like ad-hoc edditing/patching via script override via the Firefox Developer tools does not work at this time. See this related Mozilla Ideas post [here](https://connect.mozilla.org/t5/ideas/javascript-code-override-under-developer-tools/idi-p/21873/page/2#comments).

Here is a screenshot from FireFox v131:
![image](https://github.com/user-attachments/assets/cbaa5bd5-9cef-4bde-9eed-e4bc1f2a8eb9)

Here are the steps I followed:

1. I downloaded the latest xpi and extracted to a folder (xpi's are in zip format)
1. Made changes to the extension code
1. Used the **"Load Temporary Add-on..."** feature  
    It is located on the `about:debugging#/runtime/this-firefox` page.  
    Select the `mainifest.json` from the extracted folder.

Using this approach I was able to load and test changes to the extension. There might be better / more efficient way to develop extensions in real-time but this worked for this quick edit.  
**Note** that each change to the source code required me to click  **Reload** on the temporary extension.

~~One could make such changes permanent repackaging and installing the patched xpi. One would need to temporarily tell Firefox to ignore add-on signatures via `about:config` `xpinstall.signatures.required`.~~  
👆 This simple zipping unsigned repacking method should work with Development/Nightly releases of Firefox.  
Unfortunately its not that simple for the normal/mainstream releases of Firefox which block unsigned extensions, which is a good thing from an InfoSec perspective. 

In 2024-Oct I had a short email chat with the Firefox **Quick Tabs Ported** extension author about tips on how to make changes to the extension, and the above 👆 is my knowledge share. I'm assuming the author doesn't have time/interest to maintain the extension at this time.

It would still be nice to have an official **Quick Tabs** Firefox extension or at least a maintained fork but until then, I hope this info helps.

The changes I made to fix the focus issue are visible on this Gist.  The changeset is ~2 lines of real code.
https://gist.github.com/kyle0r/2dd8d30a6d1f335b68feb468293a8466/revisions#diff-3df6958c77d615b266b54fcf12fadf313e7c1e4b168f0c61ea8c3424eb39eb8c  

Here is the main gist link: https://gist.github.com/kyle0r/2dd8d30a6d1f335b68feb468293a8466 

edit: If someone is interested, I could repackage the change into an xpi and find somewhere to host it.  
edit2: Corrected inaccurate info.

**Comment by DesertBanana** [comment]
@kyle0r Thanks for your hard work! As I am non-technical person, I would greatly appreciate a repackaged and hosted xpi. Please let me know if you end up doing this.

**Comment by kyle0r** [comment]
> @kyle0r Thanks for your hard work! As I am non-technical person, I would greatly appreciate a repackaged and hosted xpi. Please let me know if you end up doing this.

No worries.

It did a bit of research on re/packaging extensions. It looks like my information about disabling extension signing only applies to non "release" versions of Firefox. So the developer and nightly builds and perhaps the ESR builds. So for "release" versions... signed extensions are required otherwise Firefox refuses to install the xpi.

Research suggests it should be _relatively_ straightforward to repackage and sign and existing extension on the "unlisted" channel. Assuming it's easy to obtain a set of Mozilla developer credentials, I will give it a go in the coming days and if I have any success I'll share the details and hopefully a signed "unlisted" xpi.

**Comment by DesertBanana** [comment]
@kyle0r Thank you!!

**Comment by kyle0r** [comment]
OK, I made and verified an AMO account (addons.mozilla.org). I was able to set up an API Key and follow the instructions on stackoverflow [here](https://stackoverflow.com/a/59172713/490487) to create and sign the attached modified XPI under a new **unlisted** extension id. 

This is an **unlisted** modification of the original [Quick Tabs Ported version 1.0.6](https://addons.mozilla.org/firefox/addon/quick-tabs-ported)

@DesertBanana @ErichDonGubler You should find a zip file link (attachment to this post). Inside the ZIP is the new xpi.  
I've have designated this version: 1.0.6.2.  
**Download link:** [quick_tabs_ported-1.0.6.2.zip](https://github.com/user-attachments/files/18262068/quick_tabs_ported-1.0.6.2.zip) (attached/hosted here on this issue thread)

ZIP archive checksums:  
SHA256: 5724961971195a572399410d2bc47688c35515bb7260b8c020c213caf04b5976  
SHA1: c56aedc7c5f3945eca3eb7468c43ae844723135d  

# Quick Tabs Ported 1.0.6.2 Changelog
* Fixed issue with search field focus when the popup is triggered.
* As suggested by a reviewer: pwrsurge [[link](https://addons.mozilla.org/firefox/addon/quick-tabs-ported/reviews/2143751/)]:
    Added option/feature **MinSearchQueryLength** which defaults to 1 which maintains the default/historic behaviour.  
    
* As suggested by reviewer: reid [[link](https://addons.mozilla.org/firefox/addon/quick-tabs-ported/reviews/2048895/)]:
    Added `<body>` margin styling to resolve cosmetic issue with the [x] close button being pushed off the right of the popup and the overflow being hidden/covered by the vertical scrollbar.

# Installation
💡 The attached extension is signed by Mozilla but is **unlisted** and has a different id/identifier to the original Quick Tabs Ported extension. This identifier mechanism prevents arbitrary changes/takeovers of an existing extension. A side effect is that the extension will be treated as a separate/different extension to the existing **listed** extension. Therefore the old/existing extension must be disabled or removed to avoid issues/conflicts/confusion.

1. Extract the xpi from the zip, note the path
1. [IF ENABLED] Open the `about:addons` page and disable to previous/old Quick Tabs Ported Add-on
1. Open the `about:addons` page and choose **Install Add-on From File...** from the gear menu
    ![image](https://github.com/user-attachments/assets/c200c390-6df3-4ee4-8951-e550e9bd58a9)
1. You will need to check / set the extension options (don't forget to click the apply button)
1. You will need to check **Manage Extension Shortcuts** from the `about:addons` gear menu to set your preferences

**Is this safe?**  
* Yes, it should be. The 1.0.6.2 xpi has been scanned, signed and approved by Mozilla via their `web-ext.js sign` tool.
* The required extension permissions are unchanged
* Normal Firefox releases will not install unsigned or corrupt extensions
* Here is the [virustotal result](https://www.virustotal.com/gui/file/5724961971195a572399410d2bc47688c35515bb7260b8c020c213caf04b5976) for the zip.
* You can use your local virus scanner to check the archive.
* You can use a tool like Beyond Compare [[link](https://www.scootersoftware.com/)] to make a diff of the xpi archives. See screenshot.
    * The signed and official Quick Tabs Ported 1.0.6 version xpi link is [here](https://addons.mozilla.org/firefox/downloads/file/3467566/quick_tabs_ported-1.0.6.xpi).

![image](https://github.com/user-attachments/assets/032a47c4-0823-4fd6-810e-de0db6171474)

## 30 second video of the changeset

https://github.com/user-attachments/assets/337725e4-f407-47ae-aafe-2776c4f3f86e

# A few notes from development

* Delete the `META-INF` folder from the original extracted XPI
* Make and test the changes to the extension
    (`about:debugging#/runtime/this-firefox` → Load Temporary Add-on... → Browse to the extracted `manifest.json`)
    Use the **reload** button on the Temporary Add-on to pick up changes
* Use `web-ext.js lint` as many times as needed to check the extension for issues prior to `web-ext.js sign`
* One-time change: in `manifest.json`
    * change the `gecko -> id` key to a new GUID [[link](https://www.uuidgenerator.net/version4)]
* For each release to be signed: increment the `version` key in `mainifest.json`
    Note: versions cannot be reused/republished.
* Follow the instructions: https://stackoverflow.com/a/59172713/490487

**Comment by ErichDonGubler** [comment]
@kyle0r: WFM now! Woot! 😀

**Comment by DesertBanana** [comment]
@kyle0r I tested it and it works perfectly, thank you so much!

**Comment by kyle0r** [comment]
Greetings in 2025,

If there any future issues with [v1.0.6.2]( https://github.com/babyman/quick-tabs-chrome-extension/issues/189#issuecomment-2563974135) of this extension feel free to write an update on this thread and tag me. I'll take a look when time allows.

As of writing I don't have time to create a longer term / auto-updating solution. Nonetheless, I think I've left enough info for someone with time and energy to continue/create a Firefox port.

Perhaps @babyman could review this comment and see if he has any thoughts / opinions, especially on the fork vs. cross-browser topic.

---

After further email conversation with author of the **Quick Tabs Ported** extension [[Firefox store link](https://addons.mozilla.org/firefox/addon/quick-tabs-ported)], I think its worth noting the following:

1. The Firefox **Quick Tabs Ported** extension author has provided some tips and suggestions and essentially said: do whatever you want. **However:** the author has explicitly stated he **does not** wish to be involved.
1. It _should_ be relatively easy to fork the HEAD of this repo and make a new Firefox port [[ref](https://extensionworkshop.com/documentation/develop/porting-a-google-chrome-extension/)]
    A future extension author would need to consider the fixes made in the signed v1.0.6.2 xpi attached previously in this thread.
1. It might even be possible to create a PR on this repo to introduce cross-browser compatibility code, which would negate the need for a Firefox specific fork/port [[ref](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Build_a_cross_browser_extension)]
    Initially, I expect this may require more code changes overall in the repo and likely introduce some conditional code. Long term this might be the _nicest_ solution but does add additional complexity like handling build+submission of the extension to both Chrome and Firefox stores.

The last two points are possible because:

> every major browser now has the same or very similar API due to the popularity of Google Chrome and its Extension API.

# Some references to wrap up

## How to sign a Firefox Extension

https://stackoverflow.com/questions/34608873/how-to-sign-a-firefox-extension  
My notes on following these steps to create v1.0.6.2 [here](https://github.com/babyman/quick-tabs-chrome-extension/issues/189#issuecomment-2563974135).

## Mozilla's Extension Workshop

https://extensionworkshop.com

Submitting an Add-on:  
https://extensionworkshop.com/documentation/publish/submitting-an-add-on/

Signing and distributing your Add-on:  
https://extensionworkshop.com/documentation/publish/signing-and-distribution-overview/

When do you need an Add-on ID?:  
https://extensionworkshop.com/documentation/develop/extensions-and-the-add-on-id/#when-do-you-need-an-add-on-id

Porting a Google Chrome extension:  
https://extensionworkshop.com/documentation/develop/porting-a-google-chrome-extension/

## The `web-ext` command reference:  

https://extensionworkshop.com/documentation/develop/web-ext-command-reference/

Some insightful extracts:

![image](https://github.com/user-attachments/assets/90febdf3-150f-4d20-a1c6-741105af371d)

`web-ext run` example under Linux:

```
# install nvm
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.1/install.sh | bash
# install node & npm
nvm install node
nvm install-latest-npm

npm i web-ext

# cd into the extracted/source dir of the extension you wish to run
~/node_modules/web-ext/bin/web-ext.js --firefox '/home/kyle/Downloads/firefox-developer-135.0b3/firefox' run

```

![image](https://github.com/user-attachments/assets/bbf6d8de-8230-445d-b06d-a135d7e9d3a9)

## Mozilla docs on Browser Extensions:

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons  
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions

manifest.json > browser_specific_settings:  
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings

### Build a cross-browser extension

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Build_a_cross_browser_extension

**Comment by DesertBanana** [comment]
@kyle0r as a happy accident, whatever you did might have also fixed a bug where the 'x' to close a tab would sometimes display right at the border of the pop up window, making it unclickable

**Comment by kyle0r** [comment]
> [@kyle0r](https://github.com/kyle0r) as a happy accident, whatever you did might have also fixed a bug where the 'x' to close a tab would sometimes display right at the border of the pop up window, making it unclickable

👆 In regards to this 👆 @DesertBanana, glad to hear the tweak helped. You can see the info in the change log in [this comment](https://github.com/babyman/quick-tabs-chrome-extension/issues/189#issuecomment-2563974135).

Here is the relevant copy/paste:

> # Quick Tabs Ported 1.0.6.2 Changelog
> 
>   * Fixed issue with search field focus when the popup is triggered.
> 
>   * As suggested by a reviewer: pwrsurge [[link](https://addons.mozilla.org/firefox/addon/quick-tabs-ported/reviews/2143751/)]:
>     Added option/feature **MinSearchQueryLength** which defaults to 1 which maintains the default/historic behaviour.
> 
>   * As suggested by reviewer: reid [[link](https://addons.mozilla.org/firefox/addon/quick-tabs-ported/reviews/2048895/)]:
>     Added `<body>` margin styling to resolve cosmetic issue with the [x] close button being pushed off the right of the popup and the overflow being hidden/covered by the vertical scrollbar.

**Comment by DesertBanana** [comment]
@kyle0r haha sorry I didn't see that earlier. Clearly I was too excited to have a working version of this extension again

**Comment by mkorthof** [comment]
@kyle0r: Thanks for your time and effort in keeping this add-on alive. just installed it and it seems to work fine 👍 Would it help to automate build/signing or even publish? Github Actions are available: 
- https://github.com/marketplace/actions/sign-firefox-xpi-file
- https://cardinalby.github.io/blog/post/github-actions/webext/6-publish-on-firefox-addons/ (part of [series](https://cardinalby.github.io/blog/series/releasing-webextension-using-github-actions/))

Was using the suggested TabSearch before, but it doesnt order tabs (last used on top) and also seems no not updated anymore.

Firefox built in tab functionality is decent, but i don't like using search bar to search my tabs.

**Comment by miaachan** [comment]
Hey everyone! 👋

Just wanted to share that I'm trying to maintain a fork of this extension for my own use, specifically focused on getting a fully functional Firefox port 🦊.

So far, I've managed to get these working:

- ✅ MRU (Most Recently Used) ordering is fully functional.
- ✅ Shortcuts for switching to older/newer tabs are working perfectly.
- ✅ All commands/shortcuts now work (like /group, /split, /merge, etc).

I used the ported version by Kevin Jones, and the enhanced version by @kyle0r for a long time (huge thanks for your work! 🙌). However, since I really missed reliable MRU ordering, snappy older/newer tab switching, and the /group functionality, I decided to fix them up and maintain a dedicated build myself.

If anyone is looking for a working Firefox version with these features restored, feel free to grab the latest signed .xpi here: 👉 https://github.com/miaachan/quick-tabs-chrome-extension/releases/tag/v2025.12.6

Hope this helps anyone else looking for a solid Quick Tabs experience on Firefox!

**Comment by DesertBanana** [comment]
> Hope this helps anyone else looking for a solid Quick Tabs experience on Firefox!

Thanks for making this. Have you noticed that Matt York's fuzzy search seems to work better than the default fuse.js search selected?

**Comment by miaachan** [comment]
> > Hope this helps anyone else looking for a solid Quick Tabs experience on Firefox!
> 
> Thanks for making this. Have you noticed that Matt York's fuzzy search seems to work better than the default fuse.js search selected?

I personally prefer /fuse since it does not require me to type everything in the exact order. My typing is sometimes just a bit too chaotic for /fuzzy lol

By the way, I just upgraded the underlying Fuse.js engine from v3 to v7. It is a pretty big update that speeds things up and finally gets rid of that old 32 character search limit. Doubt anyone is typing >32 into that box but hey it feels good to have the restriction gone

👉  https://github.com/miaachan/quick-tabs-chrome-extension/releases

**Comment by DesertBanana** [comment]
> > > Hope this helps anyone else looking for a solid Quick Tabs experience on Firefox!
> > 
> > 
> > Thanks for making this. Have you noticed that Matt York's fuzzy search seems to work better than the default fuse.js search selected?
> 
> I personally prefer /fuse since it does not require me to type everything in the exact order. My typing is sometimes just a bit too chaotic for /fuzzy lol
> 
> By the way, I just upgraded the underlying Fuse.js engine from v3 to v7. It is a pretty big update that speeds things up and finally gets rid of that old 32 character search limit. Doubt anyone is typing >32 into that box but hey it feels good to have the restriction gone
> 
> 👉 https://github.com/miaachan/quick-tabs-chrome-extension/releases

I don't know if it's placebo or maybe I don't understand how the different searches work, but for whatever reason Fuzzy seems to be a lot better at returning pages I most recently used. It could be that I'm often using this to navigate between editable documents and not just web pages, per se.

**Comment by DesertBanana** [comment]
@miaachan 

I'm not sure how to comment on your fork; anyways just wanted to say thanks and also there seems to be a minor bug where it's picking up phantom tabs that have already been closed. I haven't figured out how to reproduce it.

**Comment by DesertBanana** [comment]
On an unrelated note, does anyone know if it would be possible to port this https://github.com/mtimkovich/rip_quick_add to FF?

**Comment by countbasil** [comment]
> So far, I've managed to get these working:
>     * ✅ MRU (Most Recently Used) ordering is fully functional. 
>     * ✅ Shortcuts for switching to older/newer tabs are working perfectly.
>     * ✅ All commands/shortcuts now work (like /group, /split, /merge, etc).

@miaachan You are my hero! Thank you!! ❤️ 

Do you have a ko-fi account or similar way to support?
